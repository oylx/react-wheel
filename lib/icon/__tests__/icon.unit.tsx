import renderer from 'react-test-renderer'
import React from 'react'
import Icon from './../icon'
import {mount} from 'enzyme'

describe('icon', () => {
    it('xxx', () => {
        const json = renderer.create(<Icon name='wechat'/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick',()=>{
        let n = 1;
        const fn = ()=>{n=2};
        const component = mount(<Icon name="water" onClick={fn}/>)
        component.find('svg').simulate('click')
        expect(n).toEqual(2)
    })
})