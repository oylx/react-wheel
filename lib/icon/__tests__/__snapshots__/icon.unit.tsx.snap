// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon xxx 1`] = `
<span>
  <svg
    className="fui-icon"
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;
